### ISSUE TITLE
fix(tensorflow): safe_svd backward eye-based masking produces inf/NaN for near-equal singular values

### ISSUE THREAD

**Issue by hunter-heidenreich** [hunter-heidenreich/Kabsch-Cookbook#123]
## Problem

The masking logic in the SVD backward at `src/kabsch_horn/tensorflow/kabsch_svd_nd.py:35`:

```python
tf.where(tf.abs(S_sq_diff) < 1e-12, eye * 1e-12, S_sq_diff)
```

uses the identity matrix to fill small values. For **off-diagonal** entries where singular values are nearly equal, `eye[i,j] = 0`, so the replacement is `0 * 1e-12 = 0`. This produces `F = 1/0 = inf` on those entries, which propagates `inf` or `NaN` through the gradient.

This defeats the entire purpose of the safe SVD wrapper. The `call_safe_eigh` backward in `horn_quat_3d.py:21-22` already handles this correctly with `eps * tf.sign(D + eps)`, as do the PyTorch and JAX SVD backwards.

## Suggested Fix

Replace the eye-based masking with `eps * tf.sign(S_sq_diff + eps)`, then separately set the diagonal of `safe_D` to `1.0` and zero the diagonal of `F`, matching the pattern in the other frameworks.

Found by automated audit.
